app/models/document.rb

```diff
@@ -208,6 +214,12 @@ def save
   def publish!
     handle_remote_error do
       update_type = self.update_type || 'major'
+
+      if not_published?
+        self.change_note = "First published."
+        self.save
+      end
```

Comment:
Oh, this mysterious `ever_been_published?` method I mention is one I propose to write in this PR to do the search through change history for "First published", instead of using `not_published?` in this conditional.